### PR TITLE
qemu: update to 8.2.2

### PR DESCRIPTION
**Pull request by davidkaroly**
successful build on 64-bit and 32-bit

Smoke test on hrev57662 x86_64 host with hrev57662 guest (both 64-bit and 32-bit) was successful.
Not tried on 32-bit host.

### CODE REVIEW

File: app-emulation/qemu/qemu-8.2.2.recipe [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+	cmd:gcov$secondaryArchSuffix
+	cmd:gprof$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:ninja
+	cmd:find
+	cmd:iasl
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:pod2man
+	cmd:python3.9
```
[the remark this answers is not in the document]
Comment (davidkaroly) [reply app-emulation/qemu/qemu-8.2.2.recipe]: I had a build failure when I first tried it with `cmd:python3` - but then I was not able to reproduce the issue so probably it was only a one-time problem (I have a new laptop so new build environment). So we can go with `cmd:python3` after all.

**Comment by Begasus** [comment]
Any reason to keep the older versions around once this would be merged?

**Comment by davidkaroly** [comment]
> Any reason to keep the older versions around once this would be merged?

Not that I could think of...  I kept the old versions because this is how it was when I started contributing. But from my point of view, we can remove the old ones, anyway git remembers the commit history in case anyone needs an older recipe or patchset for reference.

**Comment by Begasus** [comment]
Can't seem to boot into a recently build image? (works with the current one)

qemu_boot_failure

**Comment by Begasus** [comment]
Booting the riscv64 image works though.

**Comment by davidkaroly** [comment]
> Can't seem to boot into a recently build image? (works with the current one)
> 

Can you share the command line your're using?

**Comment by davidkaroly** [comment]
Do you get the same issue also with OVMF or only SeaBios? Anyway, I'll try to do some testing to compare qemu 8.1 vs 8.2, also compare it on Haiku vs Fedora to see if it's even a porting issue or an upstream regression in qemu.

**Comment by Begasus** [comment]
Never tried switching BIOS in the script, maybe worth mentioning (seems I forgot), I'm still on R1B4 64bit.

**Comment by davidkaroly** [comment]
okay, upstream issue confirmed on qemu-8.2.2.
It also happens on Fedora with qemu installed from official rpm repo.

The problem seems to be: qemu is not able to boot from emulated NVMe device if memory size is set to 4G.

Boot from emulated HD or CDROM (ie. `-hda haiku-blabla.iso` or `-cdrom haiku-blabla.iso`) is OK.
Boot from emulated NVMe with 2G or 3G RAM is OK.
Boot from emulated NVMe with 4G RAM is broken.

So for the time being, perhaps you can try using `-hda` instead of `-device nvme`?
I'll also take a look at upstream qemu repo to see if there's already a ticket, or if latest `master` or `stable-8.2` is any better.

**Comment by Begasus** [comment]
Haven't figured out how to switch to using hda, but setting memory to 2GB works fine (3GB doesn't work here also), so I guess it's a known issue and this can be merged then?

**Comment by davidkaroly** [comment]
> Haven't figured out how to switch to using hda, but setting memory to 2GB works fine (3GB doesn't work here also), so I guess it's a known issue and this can be merged then?


I get a successful boot to desktop with a script like this:
```
#!sh

args=(
        -display haiku
        -machine q35
        -accel tcg
        -cpu max
        -m 4G
        -serial stdio
        -parallel none
        -monitor none
#       -bios "OVMF.fd"
        -device cirrus-vga
        -smp 8

        -device nec-usb-xhci,id=xhci
        -device usb-tablet,bus=xhci.0
        -device usb-kbd,bus=xhci.0

        -hda /boot/home/haiku-master-hrev57666-x86_64-anyboot.iso
)

qemu-system-x86_64 "${args[@]}"
```

So basically the only thing I changed was to use `-hda /path/to/haiku-image.iso` instead of the `-device nvme` and `-drive` options.


I think we should merge it, as this seems to be a known behavior in upstream qemu/SeaBIOS... (unless there's any special use case for using NVMe emulation? which I doubt)

**Comment by Begasus** [comment]
Works for me too! Thanks (thought someone would have merged already, I'm not the most experienced qemu user) :)

**Pull request merged**